### ISSUE TITLE
xeus makes use of deprecated cppzmq API (as of cppzmq 4.7)

### ISSUE THREAD

**Issue by SylvainCorlay** [jupyter-xeus/xeus#230]
The culprit is function `setsockopt`.

```
/root/xeus-python-wheel/_skbuild/linux-x86_64-3.8/cmake-build/_deps/xeus-src/src/xcontrol.cpp: In constructor ‘xeus::xcontrol::xcontrol(zmq::context_t&, const string&, const string&, const string&, xeus::xserver_zmq_split*)’:
/root/xeus-python-wheel/_skbuild/linux-x86_64-3.8/cmake-build/_deps/xeus-src/src/xcontrol.cpp:32:67: warning: ‘void zmq::detail::socket_base::setsockopt(int, const T&) [with T = int]’ is deprecated: from 4.7.0, use `set` taking option from zmq::sockopt [-Wdeprecated-declarations]
         m_publisher_pub.setsockopt(ZMQ_LINGER, get_socket_linger());
                                                                   ^
In file included from /root/xeus-python-wheel/_skbuild/linux-x86_64-3.8/cmake-build/_deps/xeus-src/include/xeus/xmiddleware.hpp:15,
                 from /root/xeus-python-wheel/_skbuild/linux-x86_64-3.8/cmake-build/_deps/xeus-src/src/xcontrol.cpp:14:
/root/xeus-python-wheel/_skbuild/linux-x86_64-3.8/cmake-build/_deps/cppzmq-src/zmq.hpp:1577:10: note: declared here
     void setsockopt(int option_, T const &optval)
          ^~~~~~~~~~
/root/xeus-python-wheel/_skbuild/linux-x86_64-3.8/cmake-build/_deps/cppzmq-src/zmq.hpp: In instantiation of ‘void zmq::detail::socket_base::setsockopt(int, const T&) [with T = int]’:
/root/xeus-python-wheel/_skbuild/linux-x86_64-3.8/cmake-build/_deps/xeus-src/src/xcontrol.cpp:32:67:   required from here
/root/xeus-python-wheel/_skbuild/linux-x86_64-3.8/cmake-build/_deps/cppzmq-src/zmq.hpp:1579:9: warning: ‘void zmq::detail::socket_base::setsockopt(int, const void*, size_t)’ is deprecated: from 4.7.0, use `set` taking option from zmq::sockopt [-Wdeprecated-declarations]
         setsockopt(option_, &optval, sizeof(T));
         ^~~~~~~~~~
/root/xeus-python-wheel/_skbuild/linux-x86_64-3.8/cmake-build/_deps/cppzmq-src/zmq.hpp:1583:10: note: declared here
     void setsockopt(int option_, const void *optval_, size_t optvallen_)
```

**Comment by SylvainCorlay** [comment]
Ping @marimeireles this may be a change you could make in core xeus!

**Comment by marimeireles** [comment]
According to @JohanMabille a good approach for this would be:
   * start with building the library, fix the warnings, rebuild, until you have no more warnings

   * when done, install it and run the tests (check travis.ci to see how to run the tests, I don't remember if you need to build the example or not)

   * when everything looks fine, try to build either xeus-python or xeus-sqlite with the installed version of xeus and check everything is fine

This method is not available on 4.6 yet, only on master, so we need to wait for a future release to update our codes.

**Comment by JohanMabille** [comment]
This has been fixed when upgrading to cppzmq 4.7.1.
